core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -162,6 +162,19 @@ class GroupMetadataManager(val brokerId: Int,
     }
   }
 
+  /**
+   * Removes all the groups and corresponding offsets for this topicPartition.
+   * @param topicPartition
+   */
+  def removeOffsetsByTopicPartition(topicPartition: TopicAndPartition): Unit = {
+    val startMs = SystemTime.milliseconds
+    val numberOfDeletedExpiredOffsets = inWriteLock(offsetExpireLock) {
+      val expireOffsets = offsetsCache.filter(_._1.topicPartition == topicPartition)
+      deleteOffsets(expireOffsets)
+    }
+    info(s"Removed ${numberOfDeletedExpiredOffsets} expired offsets in ${SystemTime.milliseconds - startMs} milliseconds for topicPartition ${topicPartition}")
```

Comment:
This method `removeOffsetsByTopicPartition` seems to be a generic method that is unrelated to expiration, it seems to delete all the offsets for the provided `topicPartition`. However, the variable names and the logging mention expired offsets. Same for the lock `offsetExpireLock`. This seems confusing. It seems to me that the refactoring didn't go far enough as we now delete offsets even if they are not expired, and we need to fix all the code to make that clear.